rush/rush/src/utilities/InstallManager.ts

```diff
@@ -46,6 +46,18 @@ export enum InstallType {
 }
 
 /**
+ * Used when determining which versions should be implicitly pinned,
+ * this interface records all version specifiers as well as whether
+ * all versions were optional or not.
+ */
+export interface IImplicitVersion {
```

Comment:
>export [](start = 0, length = 6)

Why are you exporting this?

If it's a private helper for building a data structure, then I guess I'm cool with it.

If you are really exporting it, then it should be a class with more well-defined semantics.